helper/resource/testing.go

```diff
@@ -282,8 +282,12 @@ func testStep(
 		return state, fmt.Errorf("Error on follow-up plan: %s", err)
 	}
 	if p.Diff != nil && !p.Diff.Empty() && !step.ExpectNonEmptyPlan {
```

Comment:
@stack72 caught the logic error here - `!step.ExpectNonEmptyPlan` needs to come off this line